scripts/release_info.py

```diff
@@ -119,6 +132,71 @@ def _extract_issues(logs):
     return links
 
 
+def _extract_pr_numbers(logs):
+    """Extract PR numbers out of a list of Logs
+
+    Args:
+        logs: list(Log). List of Logs to parse
+
+    Returns:
+        set(int): Set of PR numbers extracted from the log.
+    """
+    pr_numbers = PR_NUMBER_REGEX.findall(
+        ' '.join([log.message for log in logs]))
+    # Delete duplicates.
+    return list(set(pr_numbers))
+
+
+def get_prs_from_pr_numbers(pr_numbers, personal_access_token):
+    """Returns a list of PRs corresponding to the numbers provided.
+
+    Args:
+        pr_numbers: list(int). List of PR numbers.
+        personal_access_token: str. The personal access token of the user.
+
+    Returns:
+        list(github.PullRequest.PullRequest). The list of references to the PRs.
+    """
+    g = github.Github(personal_access_token)
+    repo = g.get_organization('oppia').get_repo('oppia')
+    pulls = [repo.get_pull(int(num)) for num in pr_numbers]
+    return list(pulls)
+
+
+def get_changelog_categories(pulls):
+    """Categorizes the given PRs into various changelog categories
+
+    Args:
+        pulls: list(github.PullRequest.PullRequest). The list of PRs to be
+            categorized.
+
+    Returns:
+        dict(str, list(str)). A list where the keys are the various changelog
+            labels, and the values are the titles of the PRs that fall under
+            that category.
+    """
+    result = {}
+    for pull in pulls:
+        labels = pull.labels
+        added_to_dict = False
+        formatted_title = '%s (#%d)' % (pull.title, pull.number)
+        for label in labels:
+            if 'CHANGELOG:' in label.name:
+                category = label.name[label.name.find(':') + 2:]
+                added_to_dict = True
+                if category in result.keys():
```

Comment:
Note that you can simplify the logic here and below by making `result` a collections.defaultdict(). Make sure to cast back to dict before returning the value.

Reply:
I didn't know about that! That simplifies it a lot :) Thanks!